executor/simple.go

```diff
@@ -575,31 +575,21 @@ func (e *SimpleExec) executeBegin(ctx context.Context, s *ast.BeginStmt) error {
 		}
 		if s.AsOf != nil {
 			// start transaction read only as of failed due to we set tx_read_ts before
-			if e.ctx.GetSessionVars().TxnReadTS > 0 {
+			if e.ctx.GetSessionVars().TxnReadTS.PeakTxnReadTS() > 0 {
```

Comment:
I think using the `used` flag to check rather than `readTS` is safer.

Reply:
I think `peak` is more proper here. It is used for the following case:

```sql
set transaction read only as of timestamp ts1
start transaction read only as of timestamp ts2 (expecting error)
begin ( This should be used as ts1 timestamp)
```